### DISCUSSION TITLE
stanct multiple group model function

### DISCUSSION THREAD

**Discussion by SvenjaSchwarck** [cdriveraus/ctsem#32] [General]
Hi Charles,

is there a multiple group model function for Stan code (stanct)?
I have only found the ctMultigroupFit function for OMX in the documentation. Sorry if I missed the stanct version.

Best and thanks,
Svenja

**Comment by cdriveraus** [comment]
Hi Svenja, there is no such function but with the increased flexibility of the time independent predictors, there is less real need. For every group except a first, baseline group, you can add a dummy coded covariate and allow it to influence all parameters (the default). The limitation of this approach is the random effects have the same sd and correlation structure across groups. For an example of the dummy coding and a simple function to generate it, see the example in the quick start towards bottom of https://github.com/cdriveraus/ctsem 
If the random effects are an issue, you can also specify those manually and have them influenced by covariates, it just takes a little more work. Let me know if you need more help or an example.

**Reply by SvenjaSchwarck** [reply]
Thank you for your feedback, we have decided to go with the dummy coded version for now, despite the limitations;
we have 2 groups (group_1 & group_2) and we are interested in the effect of the covariate (group_2_effect) on all paremeters of the DRIFT matrix;
I have a question regarding the interpretation of the covariate effects (output: $tipreds): do the values (mean, SD etc.) per effect of the covariate on the respective drift parameter (e.g. tip_group2_drift1) refer to the difference relative to the baseline group (in my case group 1)? Or am I interperating the $tipreds output incorrectly?

**Reply by cdriveraus** [reply]
Effects reported are the (linearised where necessary) regression coefficient of the tipred covariates (ie your dummy variables) affecting the system parameter, eg drift auto effect. For nonlinear parameters (generally, those that are constrained to be above or below zero such as drift auto effects or covariance matrix parameters, the estimate is only strictly accurate at baseline (0 on all tipreds). So, assuming your baseline group was coded with a zero and your other group with a 1, the answer to your question is yes, it is the difference relative to baseline group :)

**Reply by SvenjaSchwarck** [reply]
Thank you! :)

**Reply by SvenjaSchwarck** [reply]
Hi Charles,
Based on that, I have one more question:
We are interested in the cross-effect of phyiscal performance on cognitive performance. In total we have 180 time points. We want to know if this cross-effect is stronger in the 1st half of training (time point 1-90) or in the 2nd half of training (time point 91-180). 
What would be the better way: calculate a separate model for each training half and compare the cross-effect per model?
Or include this as a time dependent predictor?

Best and thanks,
Svenja

**Reply by cdriveraus** [reply]
The best way for this is to use the nonlinear capabilities of ctsem, you can see a small example in the CRAN vignette towards the bottom. Basically you need to make the cross effect parameter a function of some baseline parameter, plus a modification that only occurs in the 2nd half of time. The 'cleanest' way to do this is to create an extra latent process (called eg 'step2ndhalf') with all parameters and covariances fixes to zero, *except* the element of the TDPREDEFFECT matrix which should be 1. Then, include in the data a time dependent predictor named for eg 'secondhalf' which is zero except at time 90, when it is 1. This will basically make this extra latent process a step function, it will stay at zero, then shift to 1 and stay there. Then the cross effect parameter in question needs to be something like 'crossbaseline + cross2ndhalf * step2ndhalf', and you need to include 
```
PARS=c('crossbaseline', 'cross2ndhalf')
```
in your ctModel specification to tell it that these two aspects are free parameters to be estimated. Hopefully this makes sense if you also look at the example. There is a specification that will slightly faster, but basically uses undocumented hacks. If you need that we can discuss.

**Reply by SvenjaSchwarck** [reply]
thank you! I ran the model, the resulting parameters are:
td_step2ndhalf_secondhalf (-1.45), crossbaseline (-0.11) and cross2ndhalf (-1.51). I'm interested in the time course of the cross-effect physical performance on cognitive performance.
Do I understand correctly that the defined cross-effect  in question 'crossbaseline + cross2ndhalf * step2ndhalf'  becomes positive?   Accordingly, if we use the estimated mean values:  '-0.11 + -1.45 * -1.51' ~ 2,08?

**Reply by cdriveraus** [reply]
It seems like you are estimating the effect of the td predictor (td_step2ndhalf_secondhalf) on the latent 'step2ndhalf' process, *as well as* the effect of this latent process on your original process of interest. ie, you're trying to estimate two effect parameters -- one of them should be fixed (most simply to 1) for identification / scaling purposes. This kind of thing gets a little tricky to diagnose though so if I misinterpret something or you're confused, include the ctModelLatex pdf output, as well as a plot of the fit using `ctKalman(myfit, plot=T,kalmanvec='etaprior')`

**Reply by cdriveraus** [reply]
oh, but you're interpretation seems correct!
